tests/e2e/lib/virt_helpers.go

```diff
@@ -0,0 +1,417 @@
+	}
+
+	log.Println("Extracting channels...")
+	channels, ok, err := unstructured.NestedSlice(unstructuredManifest.UnstructuredContent(), "status", "channels")
+	if err != nil {
+		log.Printf("Error getting channels from packagemanifest: %v", err)
+		return "", err
+	}
+	if !ok {
+		return "", errors.New("failed to get channels list from " + name + " packagemanifest")
+	}
+	if len(channels) < 1 {
+		return "", errors.New("no channels listed in package manifest " + name)
+	}
+
+	firstChannel, ok := channels[0].(map[string]interface{})
+	if !ok {
+		return "", errors.New("failed to read first channel from package manifest " + name)
+	}
+
+	csv, ok, err := unstructured.NestedString(firstChannel, "currentCSV")
+	if err != nil {
+		return "", err
+	}
+	if !ok {
+		return "", errors.New("failed to get current CSV from " + name + " packagemanifest")
+	}
+	log.Printf("Current CSV is: %s", csv)
+
+	return csv, nil
+}
+
+// Checks the existence of the operator's target namespace
+func (v *VirtOperator) checkNamespace() bool {
+	// First check that the namespace exists
+	exists, _ := DoesNamespaceExist(v.Clientset, v.Namespace)
+	return exists
+}
+
+// Checks for the existence of the virtualization operator group
+func (v *VirtOperator) checkOperatorGroup() bool {
+	group := operatorsv1.OperatorGroup{}
+	err := v.Client.Get(context.TODO(), client.ObjectKey{Namespace: v.Namespace, Name: "kubevirt-hyperconverged-group"}, &group)
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+// Checks if there is a virtualization subscription
+func (v *VirtOperator) checkSubscription() bool {
+	subscription := operatorsv1alpha1.Subscription{}
+	err := v.Client.Get(context.TODO(), client.ObjectKey{Namespace: v.Namespace, Name: "hco-operatorhub"}, &subscription)
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+// Checks if the ClusterServiceVersion status has changed to ready
+func (v *VirtOperator) checkCsv() bool {
+	subscription, err := v.GetOperatorSubscription("down")
+	if err != nil {
+		if err.Error() == "no subscription found" {
+			return false
+		}
+	}
+
+	return subscription.CsvIsReady(v.Client)
+}
+
+// CheckHco looks for a HyperConvergedOperator and returns whether or not its
+// health status field is "healthy". Uses dynamic client to avoid uprooting lots
+// of package dependencies, which should probably be fixed later.
+func (v *VirtOperator) checkHco() bool {
+	resourceId := schema.GroupVersionResource{
+		Group:    "hco.kubevirt.io",
+		Resource: "hyperconvergeds",
+		Version:  "v1beta1",
+	}
+
+	unstructuredHco, err := v.Dynamic.Resource(resourceId).Namespace(v.Namespace).Get(context.Background(), "kubevirt-hyperconverged", v1.GetOptions{})
+	if err != nil {
+		log.Printf("Error getting HCO: %v", err)
+		return false
+	}
+
+	health, ok, err := unstructured.NestedString(unstructuredHco.UnstructuredContent(), "status", "systemHealthStatus")
+	if err != nil {
+		log.Printf("Error getting HCO health: %v", err)
+		return false
+	}
+	if !ok {
```

Comment:
4.12 jobs are failing here, maybe field does not exist there? https://github.com/kubevirt/hyperconverged-cluster-operator/commit/bab7b17a146b31ffbc4b749f6c70189245de64e8

I have 4.12 cluster, can try to debug (did not install kubevirt before)

Reply:
Yes, that was the problem. I could probably create my own check that does the same thing, but it was easier just to skip these tests on versions older than 4.13.

Reply:
just to we not forget, would create an issue to remove this code when we update CI and minimal version becomes 4.13

Reply:
Okay, if this code actually gets merged I will create an issue.